lib/service/Service.ts

```diff
@@ -133,72 +76,269 @@ class Service extends EventEmitter {
+      };
+      throw serviceError;
+    }
   }
 
   /**
    * Execute an atomic swap
+   * @throws ServiceError with status.INTERNAL code
    */
   public executeSwap = async ({ target_address, payload, identifier }: { target_address: string, payload: TokenSwapPayload, identifier: string }) => {
-    return this.raidenClient.tokenSwap(target_address, payload, identifier);
+    try {
+      return this.raidenClient.tokenSwap(target_address, payload, identifier);
+    } catch (err) {
+      const serviceError: ServiceError = {
+        code: status.INTERNAL,
+        name: 'internal',
+        message: 'internal error occured while executing swap',
+        metadata: err,
+      };
+      throw serviceError;
+    }
   }
 
   /*
    * Subscribe to incoming peer orders.
+   * @throws ServiceError with status.INTERNAL code
    */
   public subscribePeerOrders = async (callback: Function) => {
-    this.orderBook.on('peerOrder', order => callback(order));
+    try {
+      this.orderBook.on('peerOrder', order => callback(order));
+    }catch (err) {
+      const serviceError: ServiceError = {
+        code: status.INTERNAL,
+        name: 'internal',
+        message: 'internal error occured while subscribing peer orders',
+        metadata: err,
+      };
+      throw serviceError;
+    }
   }
 
   /*
    * Subscribe to executed swaps
    */
-  public subscribeSwaps = async (_callback: Function) => {};
+  public subscribeSwaps = async (_callback: Function) => {
+  }
+
+  /*
+   * Differently from LND and RAIDEN connections this function is a must for
+   * getInfo, so it catches possible errors and converts them into meaningful ServiceError
+   * then throws it to prevent startup
+   *
+   * No logging is required since exception to be thrown will be catched and logged at upstream
+   */
+  private constructOders = async(info: any) => {
+    try {
+      const pairs = await this.orderBook.getPairs();
+      info.numPeers = this.pool.peerCount;
+      info.numPairs = pairs.length;
+
+      let peerOrdersCount: number = 0;
+      let ownOrdersCount: number = 0;
+      for (const key in pairs) {
+        const pair = pairs[key];
+
+        const [peerOrders, ownOrders] = await Promise.all([
+          this.orderBook.getPeerOrders(pair.id, 0),
+          this.orderBook.getOwnOrders(pair.id, 0),
+        ]);
+
+        peerOrdersCount += Object.keys(peerOrders.buyOrders).length + Object.keys(peerOrders.sellOrders).length;
+        ownOrdersCount += Object.keys(ownOrders.buyOrders).length + Object.keys(ownOrders.sellOrders).length;
+      }
+
+      info.orders = {
+        peer: peerOrdersCount,
+        own: ownOrdersCount,
+      };
+    } catch (err) {
+      const serviceError: ServiceError = {
+        code: status.FAILED_PRECONDITION,
+        metadata: err,
+        name: 'order constructon',
+        message: 'could not construct orders',
+      };
+      throw serviceError;
+    }
+  }
+
+  private connectToRaiden = async() => {
```

Comment:
This is also not connecting to raiden, just getting info from the API, and like the other newly added private methods I think this can be dropped.

Reply:
same here :)